### PR TITLE
Add github actions for nuget

### PR DESCRIPTION
**Pull request by rmorlock**
see #2

### CODE REVIEW

File: .github/workflows/nuget_package_push_prerelease.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  pushprerelease:
+    runs-on: windows-latest
+    env:
+      ACTIONS_ALLOW_UNSECURE_COMMANDS: "true"
+    steps:
+      - name: Get current time
+        uses: srfrnk/current-time@master
+        id: current-time
+        with:
+          format: YYYYMMDDHHMMSS
```
Comment (hf-kklein) [inline .github/workflows/nuget_package_push_prerelease.yml:18]:
```suggestion
          format: YYYYMMDDHHmmss
```
See https://github.com/srfrnk/current-time#format and https://momentjs.com/docs/#/displaying/
- `MM` is month, we want `mm` = minutes
- `SS` is fractional second, we want `ss` = seconds

Both are wrong in the BO4E repo also ;)

File: .github/workflows/nuget_package_push.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 5.0.100
+      - name: Build/Check for compile errors (dotnet build)
+        run: dotnet build --configuration Release
+      - name: Run Unit Tests (dotnet test)
+        run: dotnet test --configuration Release
+      - name: Create Package (dotnet pack)
+        run: |
+          dotnet pack --configuration Release
```
Comment (hf-kklein) [inline .github/workflows/nuget_package_push.yml:25]: can we read the version (without the suffix) from a git tag?
```
dotnet pack -p:PackageVersion= read git tag here
```
(example from [official docs](https://docs.microsoft.com/de-de/dotnet/core/tools/dotnet-pack#examples))

File: .github/workflows/nuget_package_push_prerelease.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 5.0.100
+      - name: Build/Check for compile errors (dotnet build)
+        run: dotnet build --configuration Release --version-suffix "beta${{ steps.current-time.outputs.formattedTime }}"
+      - name: Run Unit Tests (dotnet test)
+        run: dotnet test --configuration Release
+      - name: Create Package (dotnet pack)
+        run: dotnet pack --configuration Release --version-suffix "beta${{ steps.current-time.outputs.formattedTime }}"
```
Comment (hf-kklein) [inline .github/workflows/nuget_package_push_prerelease.yml:29]: please read the version (without suffix) from a git tag

**Review by hf-kklein — CHANGES_REQUESTED** [review]

**Comment by hf-kklein** [comment]
For the git tag please see https://github.com/marketplace/actions/get-tag-name

File: .github/workflows/nuget_package_push_prerelease.yml (hunk 1 above)
Comment (rmorlock) [reply .github/workflows/nuget_package_push_prerelease.yml:18]: 9a3561d7157a35c8671acc27c7ced64d1dc6a528

File: .github/workflows/nuget_package_push_prerelease.yml [hunk 4]
```diff
@@ -15,18 +15,22 @@ jobs:
...
+        with:
+          tagRegex: "(?<package>.*)-(?<version>.*)"
       - name: Build/Check for compile errors (dotnet build)
-        run: dotnet build --configuration Release --version-suffix "beta${{ steps.current-time.outputs.formattedTime }}"
+        run: dotnet pack -p:PackageVersion=${{ steps.tagTimeslice.outputs.tag }}
       - name: Run Unit Tests (dotnet test)
         run: dotnet test --configuration Release
       - name: Create Package (dotnet pack)
-        run: dotnet pack --configuration Release --version-suffix "beta${{ steps.current-time.outputs.formattedTime }}"
+        run: dotnet pack -p:PackageVersion=${{ steps.tagTimeslice.outputs.tag }}
```
Comment (hf-kklein) [inline .github/workflows/nuget_package_push_prerelease.yml]: wir bruachen beides: suffix UND die package version

**Review by hf-kklein — APPROVED** [review]

**Comment by hf-kklein** [comment]
die weiteren tests passieren jetzt mit dem pre-release workflow direk tauf develop

**Pull request merged**